pkg/curl/patch/0004-HACK-manually-expand-broken-macro.patch

```diff
@@ -0,0 +1,25 @@
+From e5088b6fecc4897180f601d736c46093a834a162 Mon Sep 17 00:00:00 2001
+From: Randy Palamar <palamar@ualberta.ca>
+Date: Sun, 15 Oct 2023 13:08:51 -0600
+Subject: [PATCH] HACK: manually expand broken macro
+
+---
+ src/tool_setopt.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/tool_setopt.c b/src/tool_setopt.c
+index de3b78f..774f34d 100644
+--- a/src/tool_setopt.c
++++ b/src/tool_setopt.c
+@@ -222,7 +222,7 @@ static char *c_escape(const char *str, curl_off_t len)
+     cutoff = 3;
+   }
+ 
+-  result = curlx_dyn_addn(&escaped, STRCONST(""));
++  result = curlx_dyn_addn(&escaped, "", sizeof("") - 1);
```

Comment:
No idea why this is necessary, and only here. The `STRCONST()` macro is used all over the place but only this one seems to cause a build failure.

Maybe you can figure it out and drop this patch.

Reply:
It's because `STRCONST("")` expands to two parameters, but this doesn't work with function-like macros, since the arguments aren't expanded until after they replace the argument tokens in the body. The preprocessor just sees a 3 parameter macro called with 2 arguments.

I think the best solution is to just use object-like macros for the function name. I'll update the patches to do this.